library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,943 @@
+        'fallbackPersistence': 'fallback_persistence',
+        'ipProtocol': 'ip_protocol'
+    }
+
+    api_attributes = [
+        'address',
+        'port',
+        'advertise',
+        'description',
+        'sourcePort',
+        'translateAddress',
+        'translatePort',
+        'vlansDisabled',
+        'vlansEnabled',
+        'vlans',
+        'rules',
+        'autoLasthop',
+        'pool',
+        'lastHopPool',
+        'fallbackPersistence',
+        'ipProtocol',
+        'mask',
+        'disabled',
+        'enabled'
+    ]
+
+    returnables = [
+        'description',
+        'disabled',
+        'enabled',
+        'ip_protocol',
+        'mask',
+        'address',
+        'disabled_vlans',
+        'enabled_vlans',
+        'port',
+        'advertise',
+        'auto_lasthop',
+        'translate_address',
+        'translate_port',
+        'fallback_persistence',
+        'source_port',
+        'vlans',
+        'pool',
+        'last_hop_pool',
+        'irules',
+        'vlans_enabled',
+        'vlans_disabled'
+    ]
+
+    updatables = [
+        'description',
+        'disabled',
+        'enabled',
+        'mask',
+        'address',
+        'port',
+        'advertise',
+        'auto_lasthop',
+        'disabled_vlans',
+        'enabled_vlans',
+        'state',
+        'ip_protocol',
+        'fallback_persistence',
+        'translate_address',
+        'translate_port',
+        'source_port',
+        'vlans',
+        'pool',
+        'last_hop_pool',
+        'irules',
+        'vlans_enabled',
+        'vlans_disabled'
+    ]
+
+    @property
+    def state(self):
+        if self._values['state'] == 'enabled':
+            return 'present'
+        return self._values['state']
+
+    @property
+    def enabled(self):
+        if self._values['enabled'] is None:
+            return None
+        return True
+
+    @property
+    def disabled(self):
+        if self._values['disabled'] is None:
+            return None
+        return True
+
+
+class ApiParameters(Parameters):
+    @property
+    def irules(self):
+        if self._values['irules'] is None:
+            return []
+        return self._values['irules']
+
+
+class ModuleParameters(Parameters):
+    @property
+    def description(self):
+        if self._values['description'] is None:
+            return None
+        elif self._values['description'] in ['none', '']:
+            return ''
+        return self._values['description']
+
+    @property
+    def translate_address(self):
+        if self._values['translate_address'] is None:
+            return None
+        if self._values['translate_address']:
+            return 'enabled'
+        else:
+            return 'disabled'
+
+    @property
+    def translate_port(self):
+        if self._values['translate_port'] is None:
+            return None
+        if self._values['translate_port']:
+            return 'enabled'
+        else:
+            return 'disabled'
```

Comment:
why aren't these booleans? flatten_boolean?

Reply:
This is done the same way as in other modules:
https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_virtual_server.py#L2175

Both properties (translate_address and translate_port) are defined as booleans however, BIGIP REST API requires enabled/disabled values; therefore, corresponding parameters return enabled/disabled strings

Link to API Doc: 
https://clouddocs.f5.com/api/icontrol-rest/APIRef_tm_gtm_listener.html

Reply:
incorrect, this is the proper way to do this, change it:

https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_cgnat_lsn_pool.py#L571

Reply:
booleans in Ansible can be anything, flatten boolean is there for a reason

Reply:
Copy that - I have the change as suggested. 
This comment is addressed now.